src/Bicep.LangServer/Snippets/SnippetsProvider.cs

```diff
@@ -410,5 +446,56 @@ private IEnumerable<Snippet> GetRequiredPropertiesForObjectType(TypeSymbol typeS
                 }
             }
         }
+
+        public IEnumerable<Snippet> GetNestedResourceDeclarationSnippets(TypeSymbol typeSymbol)
+        {
+            // Leaving out the API version on this, because we expect its more common to inherit from the containing resource.
+            yield return new Snippet(@"resource ${1:Identifier} '${2:Type}' = {
+  name: $3
+  properties: {
+    $0
+  }
+}", prefix: "resource-with-defaults", detail: "Nested resource with defaults");
+
+            yield return new Snippet(@"resource ${1:Identifier} '${2:Type}' = {
+  name: $3
+  $0
+}", prefix: "resource-without-defaults", detail: "Nested resource without defaults");
+
+            string parentType = typeSymbol.Name;
+
+            if (resourceTypeToChildTypeSymbolsMap.ContainsKey(parentType))
+            {
+                foreach (TypeSymbol nestedTypeSymbol in resourceTypeToChildTypeSymbolsMap[typeSymbol.Name])
+                {
+                    resourceTypeInfoMap.TryGetValue(nestedTypeSymbol.Name, out (string prefix, string identifier, string bodyText, string description) resourceInfo);
+
+                    foreach (string nestedType in GetNestedResourceTypes(parentType, nestedTypeSymbol))
+                    {
+                        // The property "parent" is not allowed in nested resource. We'll remove the property before creating the snippet 
+                        string bodyText = ParentPropertyPattern.Replace(resourceInfo.bodyText, string.Empty);
+                        string text = LanguageConstants.ResourceKeyword + " " + resourceInfo.identifier + " '" + nestedType + "' = " + bodyText;
+
+                        yield return new Snippet(text, prefix: resourceInfo.prefix, detail: resourceInfo.description);
+                    }
+                }
+            }
+        }
+
+        // Nested resources must specify a single type segment, and optionally can specify an api version using the format "<type>@<apiVersion>"
+        // We'll remove types that exist in parent and return single type with api version 
+        private IEnumerable<string> GetNestedResourceTypes(string parentType, TypeSymbol nestedTypeSymbol)
+        {
+            if (nestedTypeSymbol is ResourceType resourceType)
+            {
+                foreach (string type in resourceType.TypeReference.Types)
+                {
+                    if (!parentType.Contains(type))
+                    {
+                        yield return type + "@" + resourceType.TypeReference.ApiVersion;
```

Comment:
Why does this method return an enumerable rather than a single value? And why do you need to check information from the parent at all? Shouldn't it just be a case of e.g. something like:

```csharp
if (nestedTypeSymbol is ResourceType resourceType)
{
    return $"{nestedTypeSymbol.Types.Last()}@{resourceType.TypeReference.ApiVersion}";
}
```

Reply:
Ah, I missed this part - nestedType.Types.Last(). Didn't think only the last one was applicable. Updated here:
https://github.com/Azure/bicep/commit/0c0c664878dbd906644c181a547c42f854f4d5f3